### PR TITLE
dr/quadrule

### PR DESCRIPTION
**Pull request by sebwolf-de**
With this PR I enable to use other quadrature rules for Dynamic Rupture computations.
1. I've added Dunavant's rule, you can now choose between Dunavant's rule and the old one (stroud) with `-DDR_QUAD_RULE={dunavant,stroud}`.
2. Dunavant's rule has order `2*CONVERGENCE_ORDER`, I've checked smaller values, but then the resolution is not good enough for our test problems.

I did not do a thorough analysis. We still need to:
1. Reevaluate Thomas' problem, where rupture stopped unexpectedly.
2. Tune the quadrature rule to find the best performance/accuracy ratio, in particular, there is a whole zoo of quadrules to consider.

### CODE REVIEW

File: cmake/process_users_input.cmake [hunk 1]
```diff
@@ -56,11 +56,16 @@ set(PRECISION_OPTIONS single double)
 set_property(CACHE PRECISION PROPERTY STRINGS ${PRECISION_OPTIONS})
 
 
-set(PLASTICITY_METHOD "nb" CACHE STRING "Dynamic rupture method: nb (nodal basis) is faster, ip (interpolation points) possibly more accurate. Recommended: nb")
+set(PLASTICITY_METHOD "nb" CACHE STRING "Plasticity method: nb (nodal basis) is faster, ip (interpolation points) possibly more accurate. Recommended: nb")
 set(PLASTICITY_OPTIONS nb ip)
 set_property(CACHE PLASTICITY_METHOD PROPERTY STRINGS ${PLASTICITY_OPTIONS})
 
 
+set(DR_QUAD_RULE "stroud" CACHE STRING "Dynamic quadrature rule")
```
Comment (Thomas-Ulrich) [inline cmake/process_users_input.cmake:64]: Fault quadrature rule
 or Dynamic Rupture quadrature rule

File: src/Solver/Interoperability.cpp [hunk 2]
```diff
@@ -309,7 +309,22 @@ void c_interoperability_report_device_memory_status() {
...
   }
 
-  double c_interoperability_M2invDiagonal(int no) {
+void c_interoperability_numberOfTriangleQuadraturePoints(int* n) {
+  *n = dr::misc::numberOfBoundaryGaussPoints;
+}
+
+void c_interoperability_triangleQuadratureRule(double* points, double* weights) {
+  constexpr size_t numberOfPoints = dr::misc::numberOfBoundaryGaussPoints;
+  auto pointsView = init::quadpoints::view::create(const_cast<real *>(init::quadpoints::Values));
```
Comment (Thomas-Ulrich) [inline src/Solver/Interoperability.cpp:318]: out of curiosity, where in the code are defined init::quadpoints::Values?

File: src/DynamicRupture/Initializers/BaseDRInitializer.cpp [hunk 3]
```diff
@@ -11,6 +10,8 @@ namespace seissol::dr::initializers {
 void BaseDRInitializer::initializeFault(seissol::initializers::DynamicRupture* dynRup,
                                         seissol::initializers::LTSTree* dynRupTree,
                                         seissol::Interoperability* eInteroperability) {
+  logInfo() << "Initialize Fault, using a quadrature rule with "
```
Comment (Thomas-Ulrich) [inline src/DynamicRupture/Initializers/BaseDRInitializer.cpp:13]: Initializing fault or Fault Initialized

**Review by Thomas-Ulrich — COMMENTED** [review]
Nice!
Are the scripts to generate the JSON files somewhere on Github?

File: src/Solver/Interoperability.cpp (hunk 2 above)
Comment (krenzland) [reply src/Solver/Interoperability.cpp:318]: They are created by yateto

File: src/DynamicRupture/FrictionLaws/LinearSlipWeakening.cpp [hunk 4]
```diff
@@ -20,11 +20,11 @@ void LinearSlipWeakeningLaw::calcStateVariableHook(
     std::array<real, misc::numPaddedPoints>& stateVariable,
     unsigned int timeIndex,
     unsigned int ltsFace) {
-  real resampledSlipRate[misc::numPaddedPoints];
+  real resampledSlipRate[misc::numPaddedPoints] __attribute__((aligned(32))){};
```
Comment (ravil-mobile) [inline src/DynamicRupture/FrictionLaws/LinearSlipWeakening.cpp:23]: alignas(ALIGNMENT)

File: src/DynamicRupture/FrictionLaws/BaseFrictionLaw.h [hunk 5]
```diff
@@ -109,7 +109,7 @@ class BaseFrictionLaw : public FrictionSolver {
     auto* qIPlus = (reinterpret_cast<QInterpolatedShapeT>(qInterpolatedPlus))[ltsFace];
     auto* qIMinus = (reinterpret_cast<QInterpolatedShapeT>(qInterpolatedMinus))[ltsFace];
 
-    FaultStresses faultStresses{};
+    FaultStresses faultStresses __attribute__((aligned(32))){};
```
Comment (ravil-mobile) [inline src/DynamicRupture/FrictionLaws/BaseFrictionLaw.h:112]: alignas(ALIGNMENT)

**Review by ravil-mobile — APPROVED** [review]

**Pull request merged**